tests/tox_env/python/test_python_api.py

```diff
@@ -65,3 +66,61 @@ def test_diff_msg_added_removed_changed() -> None:
 
 def test_diff_msg_no_diff() -> None:
     assert Python._diff_msg({}, {}) == "python "
+
+
+@pytest.mark.parametrize("ignore_base_python_conflict", [True, False])
+@pytest.mark.parametrize(
+    ("env_name", "base_python", "conflict"),
+    [
+        ("magic", ["pypy"], []),
+        ("magic", ["py39"], []),
+        (".pkg", ["py"], []),
+        ("py", ["pypy"], ["pypy"]),
+        ("cpython", ["pypy"], ["pypy"]),
+        ("pypy", ["cpython"], ["cpython"]),
+        ("pypy2", ["pypy3"], ["pypy3"]),
+        ("py3", ["py2"], ["py2"]),
+        ("py38", ["py39"], ["py39"]),
+        ("py38", ["py38", "py39"], ["py39"]),
+        ("py310", ["py38", "py39"], ["py38", "py39"]),
+        ("py3.11.1", ["py3.11.2"], ["py3.11.2"]),
+        ("py3-64", ["py3-32"], ["py3-32"]),
+        ("py310-magic", ["py39"], ["py39"]),
+    ],
+    ids=lambda a: "|".join(a) if isinstance(a, list) else str(a),
+)
+def test_base_python_matches_env_name(
+    env_name: str, base_python: List[str], conflict: List[str], ignore_base_python_conflict: bool
+) -> None:
+    if conflict:
+        if ignore_base_python_conflict:
+            result = Python._validate_base_python(env_name, base_python, ignore_base_python_conflict)
+            assert result == [env_name]
+        else:
+            msg = f"env name {env_name} conflicting with base python {conflict[0]}"
+            with pytest.raises(Fail, match=msg):
+                Python._validate_base_python(env_name, base_python, ignore_base_python_conflict)
+    else:
+        result = Python._validate_base_python(env_name, base_python, ignore_base_python_conflict)
+        assert result is base_python
```

Comment:
I am not a big fan of using `if` and `else` in tests and I would not mind to write three small tests, which would duplicate a couple of lines, but then would be easier to reason about - but that is possible a personal preference :-)

Reply:
IMHO It's a balancing act :blush: made some changes, but did not take it all the way.

Reply:
Thank you!